src/DssSpell.sol

```diff
@@ -19,42 +19,22 @@ pragma solidity 0.8.16;
 import "dss-exec-lib/DssExec.sol";
 import "dss-exec-lib/DssAction.sol";
 
-interface VatLike {
-    function Line() external view returns (uint256);
-    function file(bytes32 what, uint256 data) external;
-    function ilks(bytes32 ilk) external view returns (uint256 Art, uint256 rate, uint256 spot, uint256 line, uint256 dust);
-}
-
-interface VestLike {
-    function yank(uint256, uint256) external;
-}
-
-interface GemLike {
-    function transfer(address, uint256) external returns (bool);
+import { DssInstance, MCD } from "dss-test/MCD.sol";
+import { D3MInit, D3MCommonConfig, D3MAavePoolConfig, D3MAaveBufferPlanConfig } from "src/dependencies/dss-direct-deposit/D3MInit.sol";
+import { D3MInstance } from "src/dependencies/dss-direct-deposit/D3MInstance.sol";
+
+interface PoolConfiguratorLike {
+    function setReserveFreeze(address asset, bool freeze) external;
+    function setReserveInterestRateStrategyAddress(address asset, address newRateStrategyAddress) external;
+    function setReserveFactor(address asset, uint256 newReserveFactor) external;
 }
 
 contract DssSpellAction is DssAction {
     // Provides a descriptive tag for bot consumption
     // This should be modified weekly to provide a summary of the actions
-    // Hash: cast keccak -- "$(wget 'https://raw.githubusercontent.com/makerdao/community/b4b7c953df5d9391bfd2a2d80e973b2f74b80878/governance/votes/Executive%20vote%20-%20April%2026%2C%202023.md' -q -O - 2>/dev/null)"
+    // Hash: cast keccak -- "$(wget 'https://raw.githubusercontent.com/makerdao/community/98e98eae03662eeab0dd2092ccc7edafb2dd75d3/governance/votes/Executive%20vote%20-%20April%2028%2C%202023.md' -q -O - 2>/dev/null)"
     string public constant override description =
-        "2023-04-26 MakerDAO Executive Spell | Hash: 0xcb6486208b3cd076038392e547db55b56f4221ffd966ee1aa6573e6c86ec21d2";
-
-    uint256 internal constant FOUR_NINE_PCT_RATE = 1000000001516911765932351183;
-
-    uint256 internal constant WAD = 10 ** 18;
-
-    address internal constant PE_CONTRIBUTOR = 0x18A0609b14dB84bbcC3d911915a07CA9a28b9263;
-
-    uint256 internal constant END_VEST_TIMESTAMP = 1682899199; // Sun 30 Apr 23:59:59 UTC 2023
-
-    VatLike  internal immutable vat  = VatLike(DssExecLib.vat());
-    VestLike internal immutable vest = VestLike(DssExecLib.getChangelogAddress("MCD_VEST_MKR_TREASURY"));
-
-    // Turn office hours off
-    function officeHours() public pure override returns (bool) {
-        return false;
-    }
+        "2023-04-28 MakerDAO Executive Spell | Hash: 0xc42b83cc4f41bb759b62ac255533edba2a11195092d219b1eca54819d64069ca";
```

Comment:
Probably better to set office hours on explicitly and not rely on the default function.

Reply:
done